src/vcf/fixer.cpp

```diff
@@ -210,10 +243,139 @@ namespace ebi
             std::cerr << "DEBUG: line " << line_number << ": fixing duplicate: removing variant: "
             << std::string{line->begin(), line->end()} << std::endl;
         }
+
+        size_t Fixer::remove_duplicate_strings(const std::string &column,
+                                               const std::string &separator)
+        {
+            std::set<std::string> already_present;
+            auto is_value_duplicated = [&](const std::string &value, size_t index) -> bool {
+                bool first_appearance = already_present.insert(value).second;
+                return not first_appearance;
+            };
+            return remove_fields(column, separator, is_value_duplicated);
+        }
+
+        size_t Fixer::remove_duplicate_key_value_pairs(const std::string &column,
+                                                       const std::string &separator,
+                                                       const std::string &key_value_separator,
+                                                       const std::string &empty_value)
+        {
```

Comment:
Regarding point 2 and 3, how about using unordered_set instead of set, so that the search is done in O(1) in average case.
Though  the vector<bool> would work the same and is clearly simpler, we would only have to store the duplicate field names in the unordered_set, but in the vector all the fields would be stored.